### PR TITLE
Add `assertBusy` to make a package test more robust

### LINKED ISSUE #68043: [CI] PackageTests test90DoNotCloseStderrWhenQuiet on debian 9 on master
```
java.lang.AssertionError:
    Expected: a string containing "Failed to load settings from [elasticsearch.yml]"
         but: was "-- Logs begin at Wed 2021-01-27 09:04:57 UTC, end at Wed 2021-01-27 09:11:02 UTC. --"
        at __randomizedtesting.SeedInfo.seed([7E631C65A66D4D36:A155FE0A10CBB237]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.packaging.test.PackageTests.lambda$test90DoNotCloseStderrWhenQuiet$1(PackageTests.java:366)
        at org.elasticsearch.packaging.test.PackagingTestCase.withCustomConfig(PackagingTestCase.java:451)
        at org.elasticsearch.packaging.test.PackageTests.test90DoNotCloseStderrWhenQuiet(PackageTests.java:357)
```

Fails very rarely:
* [latest failure](https://gradle-enterprise.elastic.co/s/nlxn4z2gpwhdi)
* [all failures](https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/Bucharest&tests.container=org.elasticsearch.packaging.test.PackageTests&tests.sortField=FAILED&tests.test=test90DoNotCloseStderrWhenQuiet&tests.unstableOnly=true)

### PR DESCRIPTION
**Pull request by pugnascotia**
Closes #68043.

`PackageTests.test90DoNotCloseStderrWhenQuiet` failed because of a race
between ES startup and the test making an assertion on the ES output.
Fix this by adding a wrapper to retry the assertion for a while, until
either the assertion passes or the timeout expires.

### CODE REVIEW

File: qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java [hunk 1]
```diff
@@ -467,4 +470,44 @@ public static void setHeap(String heapSize, Path config) throws IOException {
...
+    public static void assertBusy(CheckedRunnable<Exception> codeBlock) throws Exception {
+        assertBusy(codeBlock, 10, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Runs the code block for the provided interval, waiting for no assertions to trip.
+     */
+    public static void assertBusy(CheckedRunnable<Exception> codeBlock, long maxWaitTime, TimeUnit unit) throws Exception {
+        long maxTimeInMillis = TimeUnit.MILLISECONDS.convert(maxWaitTime, unit);
+        // In case you've forgotten your high-school studies, log10(x) / log10(y) == log y(x)
```
Comment (mark-vieira) [inline qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java:486]: I definitely _have_ forgotten that.

**Review by mark-vieira — APPROVED** [review]
LGTM. Thanks, Rory!

File: qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java (hunk 1 above)
Comment (pugnascotia) [reply qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java:486]: I added the comment in `ESTestCase` because I had forgotten too 😅

**Comment by pugnascotia** [comment]
Backported to `7.x` in d21bec77b30

**Pull request merged**